### DISCUSSION TITLE
Improve the Orchard Core UI / change to a SPA?

### DISCUSSION THREAD

**Discussion by randaratceridian** [OrchardCMS/OrchardCore#10965] [Q&A]
I was wondering if there were any plans to improve the UI for editing content. Specifically:

- Right now, it's fairly simple compared to something like Umbraco which has features like storing your content in a hierarchy
- It uses a lot of full page postback when saving content.

We've written a custom layer on top of the editor, but we are considering re-writing it and converting it to a single-page application (SPA). What we hope to gain is:

- We are currently using a lot of React, so hoping to keep more in React instead of the mish mash of Razor and React we have now
- Better user experience with partial loading/saving of the page instead of full page postbacks.

 But if you had plans, we wanted to keep our plans aligned with yours.

**Comment by Skrypt** [comment]
This is not a small task to take on. I don't think using full-page postbacks is an issue. Sometimes not disposing of `HTML` `DOM` elements becomes also an issue with these client-side frameworks. Also, Razor templates with server-side controllers are pretty effective and stable. Here, I think the idea is about reliability whereas these client-side frameworks also come with a lot of dependencies and extra management of deprecated ones.

The question is always about using the right balance of both and I think a `SPA` for an admin theme might be something that should be done outside of the `OC` repository as an outside feature. Also, we need to consider `Blazor` in the equation and somehow it makes more sense than using `React` and/or `Angular` since we would stay in the same ecosystem.

**Comment by hishamco** [comment]
As @Skrypt said it's not an easy task, this could be OSS, Orchard Core Contrib (OCC) are welcome for any contribution, I also planned to add a Blazor admin in upcoming months

**Comment by ns8482e** [comment]
I agree with @Skrypt  and @hishamco  It's not easy task.   IMO Orchard core's biggest plus point is  it uses serverside rendering with it's display management, allows any developer to create Custom Admin theme and override admin templates and can also customize at run time. That I don't think is posible with SPA  like React where the templates are compiled to JS at build time and required you to work on source code.

SPA can be used  in OC's admin in different parts independently e.g OC is already using  Vue  for option, form, flow editors
